### DISCUSSION TITLE
Cinema4d color150 segmentation reference scene

### DISCUSSION THREAD

**Discussion by AugmentedRealityCat** [Mikubill/sd-webui-controlnet#445] [Show and tell]
I made a Cinema4d reference scene for all the colors referenced in the Color150 semantic segmentation system used by ControlNet. 

![Color150_semantic_segmentation_ref_v13](https://user-images.githubusercontent.com/114955776/222279603-6d2af77a-ff0c-4fc8-ab37-344d46cd8302.png)

The scene contains:

- 150 materials with the name of the class of objects they represent
- 150 text objects with those 150 materials applied to them, with proper naming and the corresponding number from 1 to 150

The text objects are all editable so you can translate these presets for other languages easily, or change the font and other parameters. You can also edit how the grid is presented using the cloner object at the top of the hierarchy.

You can also use the search function of the object manager to look for a given keyword or its corresponding number. For example, if you look for the word "screen" you'll get 3 results:

![screen_search](https://user-images.githubusercontent.com/114955776/222374081-2da7ef77-cbd5-4993-b013-295184ea8f11.png)

For reference, here is the data used to create this Cinema4d scene: 
https://docs.google.com/spreadsheets/d/1se8YEtb2detS7OuPE86fXGyD269pMycAWe2mtKUj2W8/edit#gid=0

If you have doubts about any of the color codes, names or anything, please cross-check with that spreadsheet and notice me of any error you might find so I can update the reference. 

This first public release is v13

**DOWNLOAD LINK**
[Color150_semantic_segmentation_ref_v13.zip](https://github.com/Mikubill/sd-webui-controlnet/files/10865675/Color150_semantic_segmentation_ref_v13.zip)

Your feedback is more than welcome !

This is only for the ADE20K Color150 Semantic Segmentation model as used by ControlNet - it does not work with the T2i segmentation model, which is based on a completely different set.

**Comment by Minglin-Chen** [comment]
There are two "colunm" in the reference scene.

**Reply by AugmentedRealityCat** [reply]
Thanks for spotting that one ! Here is a quick errata:

The first (red #FF0829) one is the good one for `Column/Pillar`.
The second one (pink #FF0599) is where I made a mistake, and it should have been called `Sign/Signpost` instead.
